script/api/validate_spec

```diff
@@ -1,7 +1,27 @@
 #!/usr/bin/env ruby
 
 require 'pathname'
+require 'tempfile'
+require 'psych'
+require 'yaml'
 
-path = Pathname(__dir__).join("../../docs/api/apiv3/openapi-spec.yml")
+load 'lib/api/open_api.rb'
 
-`npx @redocly/openapi-cli lint #{path}`
+assembled_path = Tempfile.new.path
+spec_path = Pathname(__dir__).join("../../docs/api/apiv3/openapi-spec.yml")
+full_spec = API::OpenAPI.send(:assemble_spec, spec_path).to_yaml
+status = 1
+
+begin
+  File.open(assembled_path, "w+") do |file|
+    file.puts full_spec
+  end
+
+  `npx @redocly/openapi-cli lint #{assembled_path}`
```

Comment:
if #10025 goes through, you will have a merge conflict here. you can add `--yes` option, with it, this script can also run for the first time on a system that does not have @redocly/openapi-cli preinstalled.

Reply:
Apparently the github action doesn't really like the `--yes` option. considered this comment invalid. ;)